test/distributed/fsdp/test_fsdp_comm_hooks.py

```diff
@@ -174,58 +222,70 @@ def test_default_communication_hook_initialization(
             Net(has_wrapping=has_wrapping, sharding_strategy=sharding_strategy),
             sharding_strategy=sharding_strategy
         )
-        dummy_state = DummyState(process_group=None)
-
-        # FSDP currently supports communication hooks for a NO_SHARD strategy
-        # Check that a `NotImplementedError` is raised for other strategies
-        if sharding_strategy != ShardingStrategy.NO_SHARD:
-            # Check that default hook is set to None
-            for entry in FSDP.fsdp_modules(fsdp_model_with_hook):
-                self.assertIsNone(entry.communication_hook)
-                self.assertIsNone(entry.communication_hook_state)
-
-            with self.assertRaisesRegex(
-                NotImplementedError,
-                '^Communication hooks are currently only available for a NO_SHARD strategy.$'
-            ):
-                fsdp_model_with_hook.register_comm_hook(dummy_state, DummyHook.dummy_hook)
 
-        else:
+        # Check that default hook is set to `all_reduce` for `NO_SHARD`
+        # or `reduce_scatter` for sharded cases
+        default_hook = default_hooks.reduce_scatter_hook\
+            if sharding_strategy != ShardingStrategy.NO_SHARD\
+            else default_hooks.allreduce_hook
+
+        for entry in FSDP.fsdp_modules(fsdp_model_with_hook):
+            self.assertEqual(entry._communication_hook, default_hook)
 
-            # Check that default hook is set to `all_reduce`
-            for entry in FSDP.fsdp_modules(fsdp_model_with_hook):
-                self.assertEqual(entry.communication_hook, default_hooks.allreduce_hook)
+        dummy_state = DummyState(process_group=None, noise=1234)
```

Comment:
Is this value of `noise` used anywhere in this test? Same with the other tests. Could you point out where the `noise` value is being used?

Reply:
It is not used. The sole purpose of the above `DummyState` is to be an example of what can be done and how a `state` can be created and how a `hook` can use it. An illustrative example was asked by Rohan offline.